wagtail/api/v2/views.py

```diff
@@ -441,7 +441,7 @@ def get_base_queryset(self):
         parent pages when using the child_of and descendant_of filters as well.
         """
         # Get live pages that are not in a private section
-        queryset = Page.objects.all().public().live()
+        queryset = self.model.objects.all().public().live()
```

Comment:
This method is used to fetch children, ancestors, etc of the page as well, would filtering this prevent those related pages from showing up if they are not the same type?

Instead, I think we could change the below like that you've added to: ``models = models_type and page_models_from_string(models_type) or [self.model]``